src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/LockStore.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls.Internal.OpenSsl
+{
+    internal unsafe static class LockStore
+    {
+        private static SemaphoreSlim[] _locks;
+        public static readonly InteropCrypto.locking_function Callback;
```

Comment:
I know it happens a lot, but I vastly prefer when people don't use "public" inside an "internal" class.  Because then you have to stop and think.  This is internal, please mark it as such.